### ISSUE TITLE
关于生成数据与原始数据一样的疑问

### ISSUE THREAD

**Issue by lyli17** [WenjieDu/SAITS#47]
您好，我在使用ett数据集进行复现时遇到了一些问题，希望能得到您的解答。
我的操作如下：
1.下载好ETTm1数据集；
2.使用python gene_ETTm1_dataset.py \--file_path RawData/ETT/ETTm1.csv \--artificial_missing_rate 0.1 \--seq_len 24 \--sliding_len 12 --dataset_name ETTm1_seqlen24_01masked \--saving_path ../generated_datasets 生成相应的数据集；
3.创建ETTm1_SAITS_base.ini并运行run_models.py进行训练；
4.python run_models.py --config_path configs/ETTm1_SAITS_best.ini --test_mode
然后将最后的插补数据.\NIPS_results\ETTm1_SAITS_best\step_196\imputations.h5转成csv文件后查看了test集里的结果跟原是真实数据一样，不知道是哪里出了问题，可否答疑一下

![Image](https://github.com/user-attachments/assets/c785600a-3b30-47cf-ad3a-67a4fce6eb1e)

**Comment by WenjieDu** [comment]
Hi there,<br><br>Thank you so much for your attention to SAITS! If you find SAITS is helpful to your work, please star⭐️ this repository. Your star is your recognition, which can let others notice SAITS. It matters and is definitely a kind of contribution.<br><br>I have received your message and will respond ASAP. Thank you again for your patience! 😃<br><br>Best,<br>Wenjie

**Comment by lyli17** [comment]
> 您好，我在使用ett数据集进行复现时遇到了一些问题，希望能得到您的解答。 我的操作如下： 1.下载好ETTm1数据集； 2.使用python gene_ETTm1_dataset.py --file_path RawData/ETT/ETTm1.csv --artificial_missing_rate 0.1 --seq_len 24 --sliding_len 12 --dataset_name ETTm1_seqlen24_01masked --saving_path ../generated_datasets 生成相应的数据集； 3.创建ETTm1_SAITS_base.ini并运行run_models.py进行训练； 4.python run_models.py --config_path configs/ETTm1_SAITS_best.ini --test_mode 然后将最后的插补数据.\NIPS_results\ETTm1_SAITS_best\step_196\imputations.h5转成csv文件后查看了test集里的结果跟原是真实数据一样，不知道是哪里出了问题，可否答疑一下
> 
> ![Image](https://github.com/user-attachments/assets/c785600a-3b30-47cf-ad3a-67a4fce6eb1e)

imputations.h5里的数据使用前面归一化采用的参数进行反归一化保存在csv里

**Comment by lyli17** [comment]
您好，可否帮忙解答一下，感觉生成的插补数据怎么跟原始数据一样，不知道哪里出了问题，但是也跑出了论文中ETT数据所对应的指标值

---- 回复的原邮件 ----
| 发件人 | Wenjie ***@***.***> |
| 发送日期 | 2025年02月10日 16:14 |
| 收件人 | WenjieDu/SAITS ***@***.***> |
| 抄送人 | lyli17 ***@***.***>,
Author ***@***.***> |
| 主题 | Re: [WenjieDu/SAITS] 关于生成数据与原始数据一样的疑问 (Issue #47) |

Hi there,

Thank you so much for your attention to SAITS! If you find SAITS is helpful to your work, please star⭐️ this repository. Your star is your recognition, which can let others notice SAITS. It matters and is definitely a kind of contribution.

I have received your message and will respond ASAP. Thank you again for your patience! 😃

Best,
Wenjie

**Comment by zhengcheng704** [comment]
> > 您好，我在使用ett数据集进行复现时遇到了一些问题，希望能得到您的解答。我的作如下： 1.下载好ETTm1数据集; 2.使用python gene_ETTm1_dataset.py --file_path RawData/ETT/ETTm1.csv --artificial_missing_rate 0.1 --seq_len 24 --sliding_len 12 --dataset_name ETTm1_seqlen24_01masked --saving_path ../generated_datasets 生成相应的数据集; 3.创建ETTm1_SAITS_base.ini并运行run_models.py进行训练; 4.python run_models.py --config_path configs/ETTm1_SAITS_best.ini --test_mode 然后将最后的插补数据.\NIPS_results\ETTm1_SAITS_best\step_196\imputations.h5转成csv文件后查看了test集里的结果跟原是真实数据一样，不知道是哪里出了问题，可否答疑一下
> > ![Image](https://github.com/user-attachments/assets/c785600a-3b30-47cf-ad3a-67a4fce6eb1e)
> 
> imputations.h5里的数据使用前面归一化采用的参数进行反归一化保存在csv里

I also encountered this issue, the test set results are the same as the original data, only the training set has imputed data.
